### PR TITLE
Add PhpStan

### PR DESCRIPTION
**Pull request by alexmanno**

### CODE REVIEW

**Comment by Algatux** [comment]
I do not agree, but I leave the decision to @ilario-pierbattista that is the new mantainer

File: .travis.yml [hunk 1]
```diff
@@ -63,6 +67,7 @@ before_script:
 script:
   - composer validate
   - bin/phpunit -c ./phpunit.xml.dist --coverage-clover=./build/coverage/coverage.clover
+  - if [[ $TEST_PHPSTAN ]]; then vendor/bin/phpstan analyse -l1 src; fi;
```
Comment (ilario-pierbattista) [inline .travis.yml]: This won't fix php version requirements as phpstan breaks dependencies installation

File: .travis.yml [hunk 2]
```diff
@@ -31,6 +31,10 @@ matrix:
     - php: '7.1'
       env:
         - SYMFONY=dev-master
+    - php: '7.1'
+      env:
+        - SYMFONY=dev-master
+        - TEST_PHPSTAN=true
```
Comment (ilario-pierbattista) [inline .travis.yml]: Same as below

File: composer.json [hunk 3]
```diff
@@ -24,7 +24,8 @@
         "matthiasnoback/symfony-dependency-injection-test": "^2.3",
         "symfony/web-profiler-bundle": "^2.8 || ^3.0 || ^4.0",
         "symfony/console": "^2.8 || ^3.0 || ^4.0",
-        "phpunit/phpunit": "^6.0"
+        "phpunit/phpunit": "^6.0",
+        "phpstan/phpstan": "^0.10.3"
```
[the remark this answers is not in the document]
Comment (Jean85) [reply composer.json]: The only way around it is to require it on-the-fly during the CI, in a step with PHP 7.1+

[the remark this answers is not in the document]
Comment (alexmanno) [reply composer.json]: @Jean85 require stuff on-the-fly IMHO is not a good idea. We can downgrade phpstan version to 0.9 and prepare a milestone for removing support to 7.0 (as a BC)

[the remark this answers is not in the document]
Comment (Jean85) [reply composer.json]: Yep, that could be a good thing. Maybe a milestone for a stable 1.0 release? @ilario-pierbattista what do you think?

[the remark this answers is not in the document]
Comment (ilario-pierbattista) [reply composer.json]: I agree, considering that php 7.0 will be no longer supported since Dec 2018.

**Comment by ilario-pierbattista** [comment]
Closing in favor of #65

**Pull request closed without merging**